Src/PackageGuard/ConfigurationLoader.cs

```diff
@@ -1,10 +1,49 @@
 using Microsoft.Extensions.Configuration;
 using PackageGuard.Core;
+using Pathy;
 
 namespace PackageGuard;
 
 public static class ConfigurationLoader
 {
+    /// <summary>
+    /// Configures the analyzer using hierarchical configuration discovery.
+    /// Looks for configuration files at solution and project levels.
+    /// </summary>
+    public static void ConfigureHierarchical(CSharpProjectAnalyzer analyzer, string projectPath)
```

Comment:
This means that this code will find all config.json in the hierarchy and merge them all in one. Instead, what I intended was that given two projects A and B, each with their own config.json, A would use its own config.json combined with the config.json at the solution level, and B uses its other config.json combined with the root config.json.

Reply:
But if the .sln file points to multiple .csproj files, they will all use the same configuration. The point of the issue was that each .csproj can have its own config.json, which combined with the global one, will be used for the analysis of that specific project.